pkg/vm/memory/memory_value.go

```diff
@@ -194,19 +183,65 @@ func (mv *MemoryValue) Equal(other *MemoryValue) bool {
 	return false
 }
 
-// Adds two memory values is the second one is a Felt
+// // Adds two memory values is the second one is a Felt
+// func (mv *MemoryValue) Add(lhs, rhs *MemoryValue) (*MemoryValue, error) {
+// 	var err error
+// 	if lhs.IsAddress() {
+// 		if !rhs.IsFelt() {
+// 			return nil, errors.New("rhs is not a felt")
+// 		}
+// 		// lhs : Address, rhs : Address
+// 		mv.address, err = mv.address.Add(lhs.address, rhs.felt)
+// 	} else {
+// 		if rhs.IsAddress() {
+// 			mv.address, err = mv.address.Add(rhs.address, lhs.felt)
+// 		} else {
+// 			mv.felt = mv.felt.Add(lhs.felt, rhs.felt)
+// 		}
+// 	}
+
+// 	if err != nil {
+// 		return nil, err
+// 	}
+// 	return mv, nil
+// }
+
+// Adds a memory address and a field element
+func (address *MemoryAddress) Add(lhs *MemoryAddress, rhs *f.Element) (*MemoryAddress, error) {
+	if !rhs.IsUint64() {
+		return nil, fmt.Errorf("field element does not fit in uint64: %s", rhs.String())
+	}
+
+	address.SegmentIndex = lhs.SegmentIndex
+	address.Offset = lhs.Offset + rhs.Uint64()
+	return address, nil
+}
```

Comment:
Why did you change the position of this function definition?

Reply:
I will change it back to where it was. I was carefully checking if there's any missing cases.